src/analysis/processing/qgsalgorithmrectanglesovalsdiamonds.cpp

```diff
@@ -0,0 +1,209 @@
+}
+
+QString QgsRectanglesOvalsDiamondsAlgorithm::group() const
+{
+  return QObject::tr( "Vector geometry" );
+}
+
+QString QgsRectanglesOvalsDiamondsAlgorithm::groupId() const
+{
+  return QStringLiteral( "vectorgeometry" );
+}
+
+QString QgsRectanglesOvalsDiamondsAlgorithm::shortHelpString() const
+{
+  return QObject::tr( "Creates a rectangle, oval or diamond-shaped polygons from the input point layer using "
+                      "specified width, height and (optional) rotation values." );
+}
+
+QString QgsRectanglesOvalsDiamondsAlgorithm::outputName() const
+{
+  return QObject::tr( "Polygon" );
+}
+
+QList<int> QgsRectanglesOvalsDiamondsAlgorithm::inputLayerTypes() const
+{
+  return QList<int>() << QgsProcessing::TypeVectorPoint;
+}
+
+QgsProcessing::SourceType QgsRectanglesOvalsDiamondsAlgorithm::outputLayerType() const
+{
+  return QgsProcessing::TypeVectorPolygon;
+}
+
+QgsWkbTypes::Type QgsRectanglesOvalsDiamondsAlgorithm::outputWkbType( QgsWkbTypes::Type inputWkbType ) const
+{
+  QgsWkbTypes::Type outputWkbType = QgsWkbTypes::Polygon;
+  if ( QgsWkbTypes::hasM( inputWkbType ) )
+  {
+    outputWkbType = QgsWkbTypes::addM( outputWkbType );
+  }
+  if ( QgsWkbTypes::hasZ( inputWkbType ) )
+  {
+    outputWkbType = QgsWkbTypes::addZ( outputWkbType );
+  }
+
+  return outputWkbType;
+}
+
+QgsRectanglesOvalsDiamondsAlgorithm *QgsRectanglesOvalsDiamondsAlgorithm::createInstance() const
+{
+  return new QgsRectanglesOvalsDiamondsAlgorithm();
+}
+
+void QgsRectanglesOvalsDiamondsAlgorithm::initParameters( const QVariantMap & )
+{
+  addParameter( new QgsProcessingParameterEnum( QStringLiteral( "SHAPE" ), QObject::tr( "Shape" ), QStringList() << QObject::tr( "Rectangle" ) << QObject::tr( "Diamond" ) << QObject::tr( "Oval" ), false, 0 ) );
+
+  auto widthParam = qgis::make_unique < QgsProcessingParameterDistance >( QStringLiteral( "WIDTH" ), QObject::tr( "Width" ), 1.0, QStringLiteral( "INPUT" ), false, 0.0 );
+  widthParam->setIsDynamic( true );
+  widthParam->setDynamicPropertyDefinition( QgsPropertyDefinition( QStringLiteral( "Width" ), QObject::tr( "Width" ), QgsPropertyDefinition::Double ) );
+  widthParam->setDynamicLayerParameterName( QStringLiteral( "INPUT" ) );
+  addParameter( widthParam.release() );
+
+  auto heightParam = qgis::make_unique < QgsProcessingParameterDistance >( QStringLiteral( "HEIGHT" ), QObject::tr( "Height" ), 1.0, QStringLiteral( "INPUT" ), false, 0.0 );
+  heightParam->setIsDynamic( true );
+  heightParam->setDynamicPropertyDefinition( QgsPropertyDefinition( QStringLiteral( "Height" ), QObject::tr( "Height" ), QgsPropertyDefinition::Double ) );
+  heightParam->setDynamicLayerParameterName( QStringLiteral( "INPUT" ) );
+  addParameter( heightParam.release() );
+
+  auto rotationParam = qgis::make_unique < QgsProcessingParameterNumber >( QStringLiteral( "ROTATION" ), QObject::tr( "Rotation" ), QgsProcessingParameterNumber::Double, 0.0, true, 0.0, 360.0 );
```

Comment:
```suggestion
  auto rotationParam = qgis::make_unique < QgsProcessingParameterNumber >( QStringLiteral( "ROTATION" ), QObject::tr( "Rotation" ), QgsProcessingParameterNumber::Double, 0.0, true, -360.0, 360.0 );
```

(just in case someone would prefer to enter -45?)

Reply:
Good catch. Changed